similar/arch/ogl/ogl.cpp

```diff
@@ -1283,6 +1290,65 @@ void ogl_start_frame(grs_canvas &canvas)
 #endif
 	glMatrixMode(GL_MODELVIEW);
 	glLoadIdentity();//clear matrix
+
+	if (!VR_stereo)
+		return;
+
+	// cache viewport + projection transform for left/right stereo views
+	glGetFloatv(GL_PROJECTION_MATRIX, &ogl_stereo_transform[0][0]);
+	glGetFloatv(GL_PROJECTION_MATRIX, &ogl_stereo_transform[1][0]);
+
+	glGetIntegerv(GL_VIEWPORT, &ogl_stereo_viewport[0][0]);
+	glGetIntegerv(GL_VIEWPORT, &ogl_stereo_viewport[1][0]);
+
+	// query if stereo quad buffering available?
+	glGetBooleanv(GL_STEREO, &ogl_stereo_enabled);
+
+	// half-width viewports for side-by-side format
+	if (VR_half_width)
+		ogl_stereo_viewport[1][0] += ogl_stereo_viewport[0][2];		// x = w/2
+	// half-height viewports for above/below format
+	else if (VR_half_height)
+		ogl_stereo_viewport[1][1] -= ogl_stereo_viewport[0][3];		// y = h/2
+
+	// center unsqueezed side-by-side format
+	if (VR_half_width && VR_half_height) {
+		ogl_stereo_viewport[0][1] -= ogl_stereo_viewport[0][3]/2;	// y = h/4
+		ogl_stereo_viewport[1][1] -= ogl_stereo_viewport[1][3]/2;	// y = h/4
+	}
+}
+
+void ogl_stereo_frame(int xeye, int xoff)
+{
+	float dxoff = xoff * 2.0f / grd_curscreen->sc_canvas.cv_bitmap.bm_w;
+
+	if (!VR_stereo)
+		return;
+
+	if (xeye < 0) {
+		// left eye view
```

Comment:
The point of the ogl_stereo_frame() function itself was intended as adjunct to ogl_start_frame(), which is why the left/right eye logic is conditionally handled here. Was not sure originally if it was going to be called more than once per frame, which is why the cached values are saved per eye. Now evident it always gets called once per frame, so the caching stuff could be compacted or eliminated.

Reply:
Relocated all stereo handling logic inside ogl_stereo_frame() since it is only ever called once per frame for the selected left or right eye. This should leave ogl_start_frame() intact with its original setup code for default non-stereo instances.